src/sensors/lorreconstruction.rs

```diff
@@ -306,11 +314,26 @@ mod test_electronics {
         assert!(lor.is_some());
     }
 
-    // #[test]
-    // fn test_weighted_mean() {
-    //     let (qs, xyz, _) = qt_vector();
+    use geometry::assert_uom_eq;
+    use uom::si::length::millimeter;
+    #[rstest]
+    fn test_length_weighted_mean(qt_vector: ([u32; 10], SensorMap, Vec<MCQT>)) {
+        let (qs, xyz, _) = qt_vector;
+        let mut xs = Vec::with_capacity(qs.len());
+        let mut qtest = Vec::with_capacity(qs.len());
+        let mut xq_sum = Length::ZERO * Ratio::ZERO;
```

Comment:
Surely `Length::ZERO * Ratio::ZERO` is just a long-winded spelling of `Length::ZERO`. Or are you trying to emphazise that it corresponds to something calculated as a product of these two quantities?

Reply:
The test is a bit old maybe. I was trying to be completely type specific since at the time I thought that `Ratio` should be treated in the same way as other units so the `sum(x*w)` wouldn't be `Length`. It's probably unnecessary though, yes.